Výstup nyní vypadá takto:
```
Podúloha 1: Tvoje řešítko spadlo... :( Program skončil s návratovým kódem 1
Před ukončením Tvůj program vypsal následující výstup:
--------- Standardní výstup Tvého programu ---------
b'43\n'
--------- Konec standardního výstupu ---------
--------- Chybový výstup Tvého programu ---------
b'44\nTraceback (most recent call last):\n  File "/home/michal/ksp-klient/python-test.py", line 8, in <module>\n    print(4 / 0)\nZeroDivisionError: division by zero\n'
--------- Konec chybového výstupu ---------
Podúloha 2: Tvoje řešítko spadlo... :( Program skončil s návratovým kódem 1
Před ukončením Tvůj program vypsal následující výstup:
--------- Standardní výstup Tvého programu ---------
b'43\n'
--------- Konec standardního výstupu ---------
--------- Chybový výstup Tvého programu ---------
b'44\nTraceback (most recent call last):\n  File "/home/michal/ksp-klient/python-test.py", line 8, in <module>\n    print(4 / 0)\nZeroDivisionError: division by zero\n'
--------- Konec chybového výstupu ---------
...
```

Nějaké návrhy na zlepšení nebo myslíte, že je ok?

--- a/ksp-klient.py
+++ b/ksp-klient.py
@@ -50,7 +50,8 @@ def error(*args, **kvargs):
     if sys.stderr.isatty():
         eprint("\033[31m", end="")
         color_end = "\033[0m"
-    eprint(*args, color_end, **kvargs)
+    eprint(*args, **kvargs)
+    eprint(color_end, end="")
 
 
 def print_nice_json(json_text):

--- a/ksp-klient.py
+++ b/ksp-klient.py
@@ -309,6 +309,13 @@ def handle_generate(arguments: Namespace) -> None:
             arguments.output_file.write(chunk)
 
 
+def try_decode_bytes(data: Union[bytes, bytearray]) -> Optional[str]:
+    try:
+        return data.decode()
+    except UnicodeDecodeError:
+        return None
+
+
 def handle_run(arguments: Namespace) -> None:
     task = arguments.task
     subtasks = arguments.subtasks
@@ -333,12 +340,12 @@ def handle_run(arguments: Namespace) -> None:
                 error("Před ukončením Tvůj program vypsal následující výstup:")
                 if e.stdout:
                     error("--------- Standardní výstup Tvého programu ---------")
-                    error(e.stdout)
-                    error("--------- Konec standardního výstupu ---------")
+                    error(try_decode_bytes(e.stdout) or e.stdout)
+                    error("--------- Konec standardního výstupu ---------------")
                 if e.stderr:
-                    error("--------- Chybový výstup Tvého programu ---------")
-                    error(e.stderr)
-                    error("--------- Konec chybového výstupu ---------")
+                    error("--------- Chybový výstup Tvého programu ------------")
+                    error(try_decode_bytes(e.stderr) or e.stderr)
+                    error("--------- Konec chybového výstupu ------------------")
         finally:
             file.close()
 

--- a/ksp-klient.py
+++ b/ksp-klient.py
@@ -340,12 +340,16 @@ def handle_run(arguments: Namespace) -> None:
                 error("Před ukončením Tvůj program vypsal následující výstup:")
                 if e.stdout:
                     error("--------- Standardní výstup Tvého programu ---------")
-                    error(try_decode_bytes(e.stdout) or e.stdout)
+                    enconded_bytes = try_decode_bytes(e.stdout)
+                    params = {} if enconded_bytes == None else {"end": ""}
+                    error(enconded_bytes or e.stdout, **params)
                     error("--------- Konec standardního výstupu ---------------")
                 if e.stderr:
                     error("--------- Chybový výstup Tvého programu ------------")
-                    error(try_decode_bytes(e.stderr) or e.stderr)
-                    error("--------- Konec chybového výstupu ------------------")
+                    enconded_bytes = try_decode_bytes(e.stderr)
+                    params = {} if enconded_bytes == None else {"end": ""}
+                    error(enconded_bytes or e.stderr, **params)
+                    error("--------- Konec chybového výstupu ---------")
         finally:
             file.close()
 

